Fixed some parsing misses

scudmarx opened the pull request:
"upgrade all weapons" didn't work because nothing is equipped with something called "weapon". Now matches any equipment against generic words.  "2x Burst Carbines" wasn't matching against "Burst Carbine" in DisabledUpgradeGroup check, now does (the count stripping regex was applied in the same declaration and the check was made against the original label rather than the stripped one.  I apply the regex strip to that label when checking for equipment availability).

AdamLay commented:
As discussed - happy to merge this if we can get some tests covering compareEquipmentNames, thank you! :)

scudmarx commented:
I had to go too far hey?

A lot of tests were failing just purely because they asserted StrictEqualTo and the returned objects had randomly generated id fields, so I changed the tests to MatchObject which passes. Fixed some 'problems' by adding a DefaultEquipment template on some test comparison objects.

And then I finally added unit tests for the compareEquipmentNames function like you asked. 😶 

Some tests are still failing, but I think they are an actual, ya know, failure, like tests are supposed to sometimes be? UpgradeService.isValid is having issues at the moment.

scudmarx commented:
Fixed the isValid chain, which turned out to be findUpgrade searching by name instead of label.

![Victory](https://user-images.githubusercontent.com/23005404/143244962-29593b14-2220-4e1d-8667-7d18fb3a7665.png)

AdamLay commented:
Excellent, looking good now - thanks!
